ci_scripts/assert_news.py

```diff
@@ -1,86 +1,153 @@
+            )
+        logger.info(
+            f':: Looking for news files in `{NEWS_DIR}` [{self.current_branch}]'
+        )
+        self.git.checkout(self.current_branch)
+        self.git.set_upstream_branch(self.current_branch)
+        self.git.pull()
+        current_commit_hash = self.git.get_commit_hash()
+        # Delete `master` as it may already exist and be set to the one of interest
+        if self.git.branch_exists(self.master_branch):
+            self.git.delete_branch(self.master_branch)
+        self.git.checkout(self.master_branch)
+        self.git.set_upstream_branch(self.master_branch)
+        self.git.pull()
+        master_branch_commit_hash = self.git.get_commit_hash()
+        self.git.checkout(self.current_branch)
+        added_news = self.git.get_changes_list(
+            'a', self.git.get_branch_point(
+                master_branch_commit_hash, current_commit_hash),
+            current_commit_hash,
+            os.path.relpath(NEWS_DIR,
+                            os.path.commonprefix([NEWS_DIR, PROJECT_ROOT]))
+        )
+        extension_to_exclude = ['.toml', '.rst']
+        return [path for path in added_news if
+                len([ex for ex in extension_to_exclude if ex in path]) == 0]
+
+    def verify(self):
+        """
+        Checks that news files were added in the current branch as part of the PR's changes
+        The files are then individually checked in order to ensure they follow the standard in terms of naming and content.
+        """
+        if self.current_branch in [MASTER_BRANCH, BETA_BRANCH] or re.search(
+                RELEASE_BRANCH_PATTERN, self.current_branch):
+            logger.info(
+                f'No need for news file on branch [{self.current_branch}]'
+            )
+            return
+        added_news = self.find_news_file()
+        if not added_news or len(added_news) == 0:
+            FileNotFoundError(
+                f'PR must contain a news file in {NEWS_DIR}. See README.md'
+            )
+        logger.info(
+            f'{len(added_news)} new news files found in `{NEWS_DIR}`'
+        )
+        logger.info(':: Checking news files format')
+        for news_file in added_news:
+            NewsFileValidator(news_file).verify()
+
+
+def main():
+    """
+    Asserts that the new PR comprises at least one news file and that such file is correct with regards to the standard.
+    An exception is raised if a problem with the news file was found.
+    """
+    parser = argparse.ArgumentParser(
+        description='Publish target data report to AWS.')
+    parser.add_argument("-v", "--verbose", action="count", default=0,
+                        help="Verbosity, by default errors are reported.")
+    args = parser.parse_args()
+    set_log_level(args.verbose)
+    try:
+        NewsFileDiscoverer().verify()
+    except Exception as e:
+        log_exception(logger, e)
+        sys.exit(1)

```

Comment:
It would be better to call sys.exit from where main is called, this may have undesired effects when called from elsewhere e.g. in tests
```suggestion
        return 1
    else:
        return 0
```